src/main/java/uk/gov/hmcts/darts/datamanagement/service/impl/InboundToUnstructuredProcessorImpl.java

```diff
@@ -53,27 +48,49 @@ public class InboundToUnstructuredProcessorImpl implements InboundToUnstructured
             FAILURE_EMPTY_FILE.getId()
         );
 
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final InboundToUnstructuredProcessorSingleElement singleElementProcessor;
+
+    @Value("${darts.automated.task.inbound-to-unstructured.threads:20}")
+    private final Integer threads;
+
+
     @Override
-    public void processInboundToUnstructured() {
+    public void processInboundToUnstructured(int batchSize) {
         log.debug("Processing Inbound data store");
-        processAllStoredInboundExternalObjectsOneCall();
-    }
-
-    private void processAllStoredInboundExternalObjectsOneCall() {
         List<ExternalObjectDirectoryEntity> inboundList = externalObjectDirectoryRepository.findEodsForTransfer(getStatus(STORED), getType(INBOUND),
                                                                                                                 getStatus(STORED), getType(UNSTRUCTURED), 3,
-                                                                                                                limit);
-        int count = 1;
-        for (ExternalObjectDirectoryEntity inboundObject : inboundList) {
-            log.debug("Processing Inbound to Unstructured record {} of {} with EOD {}", count++, inboundList.size(), inboundObject);
-            try {
-                singleElementProcessor.processSingleElement(inboundObject);
-            } catch (Exception exception) {
-                log.error("Failed to move from inbound file to unstructured data store for EOD id: {}", inboundObject.getId(), exception);
+                                                                                                                batchSize);
+        AtomicInteger count = new AtomicInteger();
+
+        List<Callable<Void>> tasks = inboundList.stream()
+            .map(inboundObject -> (Callable<Void>) () -> {
+                log.debug("Processing Inbound to Unstructured record {} of {} with EOD {}",
+                          count.getAndIncrement(), inboundList.size(), inboundObject);
+
+                processInboundToUnstructured(inboundObject);
+                return null;
+            }).toList();
+        try {
+            AsyncUtil.invokeAllAwaitTermination(tasks, threads, 1, TimeUnit.HOURS);
```

Comment:
Actually just thought if the batch size gets set to less than thread size you don't want it to wait for more threads than in the batch. Also if the number of objects returned is less than thread size its the same

Reply:
Threads are just the number of threads that get spun up that the tasks can execute on.
The await termination waits for all tasks to completed not the threads so even if we have fewer tasks then we do threads, we should not get stuck